src/api/app/services/scm_exception_handler.rb

```diff
@@ -0,0 +1,38 @@
+class SCMExceptionHandler
+  include ActiveSupport::Rescuable
+
+  rescue_from Octokit::AbuseDetected,
+              Octokit::AccountSuspended,
+              Octokit::BillingIssue,
+              Octokit::BranchNotProtected,
+              Octokit::Conflict,
+              Octokit::Forbidden,
+              Octokit::RepositoryUnavailable,
+              Octokit::NotFound,
+              Octokit::OneTimePasswordRequired,
+              Octokit::Unauthorized,
+              Octokit::UnavailableForLegalReasons,
+              Octokit::UnsupportedMediaType,
+              Octokit::CommitIsNotPartOfPullRequest,
+              Octokit::InstallationSuspended,
+              Octokit::SAMLProtected,
+              Octokit::TooLargeContent,
+              Octokit::TooManyLoginAttempts,
+              Octokit::UnverifiedEmail,
+              Octokit::InvalidRepository,
+              Octokit::PathDiffTooLarge,
+              Octokit::ServiceUnavailable,
+              Octokit::InternalServerError do |exception|
+    # FIXME: Inform users about the exceptions
+  end
+
+  rescue_from Gitlab::Error::Conflict,
+              Gitlab::Error::Forbidden,
+              Gitlab::Error::InternalServerError,
+              Gitlab::Error::MissingCredentials,
+              Gitlab::Error::NotFound,
+              Gitlab::Error::ServiceUnavailable,
+              Gitlab::Error::Unauthorized do |exception|
```

Comment:
I think we should add `Gitlab::Error::TooManyRequests` to this list if somehow the user gets rate-limited. They could always try later. It's quite unlikely, but who knows...